src/services/storage/storage-factories.ts

```diff
@@ -0,0 +1,17 @@
+import { networks } from '../../constants/networks'
+import { Network } from '../../model/storage/network.model'
+import { Swarm } from '../../model/storage/swarm.model'
+
+export function networkFactory(): Network {
+  return Object.assign({}, networks[0])
+}
+
+export function networkListFactory(): Network[] {
+  return Object.assign([], networks)
+}
+
+export function swarmFactory(): Swarm {
+  return {
+    extensionId: process.env.SWARM_EXTENSION_ID,
```

Comment:
what if `rocess.env.SWARM_EXTENSION_ID` is not defined? there should be an error handling.

Reply:
or is it defined by webpack?

Reply:
It is defined in the `.env` file. An error could be thrown here or in the place where communication is established with the Swarm extension. Probably it makes more sense to throw an error there.